src/shims/unix/foreign_items.rs

```diff
@@ -590,6 +590,13 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                     let einval = this.eval_libc("EINVAL");
                     this.set_last_error(einval)?;
                     this.write_scalar(Scalar::from_i32(-1), dest)?;
+                } else if pid != 0 {
+                    // TODO: when https://github.com/rust-lang/miri/issues/3730 is fixed this should use its notion of tid/pid
+
+                    // The thread whose ID is pid could not be found
+                    let einval = this.eval_libc("ESRCH");
+                    this.set_last_error(einval)?;
+                    this.write_scalar(Scalar::from_i32(-1), dest)?;
```

Comment:
This looks like the program did something wrong, which is not the case. Also you still have the `ThreadId::from(pid)` above which is a conversion that makes no sense.

Instead of this, please make the `_` arm in the `match pid` above do something like
```rust
throw_unsup_format!("`sched_getaffinity` is only supported with a pid of 0 (indicating the current thread)");
```